src/geo/projection/globe.js

```diff
@@ -0,0 +1,506 @@
+    const posMatrix = mat4.identity(new Float64Array(16));
+    mat4.translate(posMatrix, posMatrix, [offset[0], offset[1], -wsRadius]);
+    mat4.scale(posMatrix, posMatrix, [scale, scale, scale]);
+    mat4.rotateX(posMatrix, posMatrix, degToRad(-tr._center.lat));
+    mat4.rotateY(posMatrix, posMatrix, degToRad(-tr._center.lng));
+
+    return posMatrix;
+}
+
+export function calculateGlobeMercatorMatrix(tr: Transform): mat4 {
+    const worldSize = tr.worldSize;
+    const lat = clamp(tr.center.lat, -MAX_MERCATOR_LATITUDE, MAX_MERCATOR_LATITUDE);
+    const point = new Point(
+        mercatorXfromLng(tr.center.lng) * worldSize,
+        mercatorYfromLat(lat) * worldSize);
+
+    const mercatorZ = mercatorZfromAltitude(1, tr.center.lat) * worldSize;
+    const projectionScaler = mercatorZ / tr.pixelsPerMeter;
+    const zScale = tr.pixelsPerMeter;
+    const ws = worldSize / projectionScaler;
+
+    const posMatrix = mat4.identity(new Float64Array(16));
+    mat4.translate(posMatrix, posMatrix, [point.x, point.y, 0.0]);
+    mat4.scale(posMatrix, posMatrix, [ws, ws, zScale]);
+
+    return posMatrix;
+}
+
+export const GLOBE_ZOOM_THRESHOLD_MIN = 5;
+export const GLOBE_ZOOM_THRESHOLD_MAX = 6;
+
+export function globeToMercatorTransition(zoom: number): number {
+    return smoothstep(GLOBE_ZOOM_THRESHOLD_MIN, GLOBE_ZOOM_THRESHOLD_MAX, zoom);
+}
+
+export function globeBuffersForTileMesh(painter: Painter, tile: Tile, coord: OverscaledTileID, tiles: number): [VertexBuffer, VertexBuffer] {
+    const context = painter.context;
+    const id = coord.canonical;
+    const tr = painter.transform;
+    let gridBuffer = tile.globeGridBuffer;
+    let poleBuffer = tile.globePoleBuffer;
+
+    if (!gridBuffer) {
+        const gridMesh = GlobeSharedBuffers.createGridVertices(id.x, id.y, id.z);
+        gridBuffer = tile.globeGridBuffer = context.createVertexBuffer(gridMesh, globeLayoutAttributes, false);
+    }
+
+    if (!poleBuffer) {
+        const poleMesh = GlobeSharedBuffers.createPoleTriangleVertices(tiles, tr.tileSize * tiles, coord.canonical.y === 0);
+        poleBuffer = tile.globePoleBuffer = context.createVertexBuffer(poleMesh, globeLayoutAttributes, false);
+    }
+
+    return [gridBuffer, poleBuffer];
+}
+
+export function globeMatrixForTile(id: CanonicalTileID, globeMatrix: mat4) {
+    const gridTileId = new CanonicalTileID(id.z, Math.pow(2, id.z) / 2, id.y);
+    const bounds = globeTileBounds(gridTileId);
+    const decode = globeDenormalizeECEF(bounds);
+    const posMatrix = mat4.clone(globeMatrix);
+    mat4.mul(posMatrix, posMatrix, decode);
+
+    return posMatrix;
+}
+
+export function globeUpVectorMatrix(id: CanonicalTileID, tiles: number) {
+    // Tile up vectors and can be reused for each tiles on the same x-row.
+    // i.e. for each tile id (x, y, z) use pregenerated mesh of (0, y, z).
+    // For this reason the up vectors are rotated first by 'yRotation' to
+    // place them in the correct longitude location.
```

Comment:
Is still the case?

Reply:
No, simplified in a73e2dce39bfb25ec09953468cca2b56f607a090 and https://github.com/mapbox/mapbox-gl-js/pull/11329/commits/270bc9848dc754fab297c3e24bfec89ad663094e